packages/angular-query-experimental/src/infinite-query-options.ts

```diff
@@ -13,7 +13,7 @@ export type UndefinedInitialDataInfiniteOptions<
   TQueryFnData,
   TError = DefaultError,
   TData = InfiniteData<TQueryFnData>,
-  TQueryKey extends QueryKey = QueryKey,
+  TQueryKey extends QueryKey = Array<any>,
```

Comment:
my idea was to keep this to:

```
TQueryKey extends QueryKey = QueryKey,
```

and then just change what `QueryKey` is:

```diff
- export type QueryKey = ReadonlyArray<unknown>
+ export type QueryKey = Array<any>
```

Reply:
Right! Sorry my bad, you did write that. Have tried it just to double check but no dice. The initial error about it being readonly is part of the problem, but the full problem is the flipped order of assigning sub-type to super-type caused by the type being referenced in a callback. 

<img width="583" alt="image" src="https://github.com/user-attachments/assets/64d57ae5-ad83-4898-a3fb-58048c4d8cee">

The fix for this is always to provide your types correctly at the top layer so both directions of structural comparison can pass, and Tanstack Query has been masking this problem through query keys not always being respected on Client methods. This masking became really obvious with the new tRPC Client though because we can't type-anything without the DataTag flowing through

Reply:
Wait I made another mistake there keeping unknown, will come back

Reply:
Okay yep same answer as above:

`Array<unknown>` -> we get assignment error in callbacks

`Array<any>` -> assignment errors fixed, but we instead of get errors when assigning const/inline keys to query keys (can't assign readonly to mutable)

`ReadonlyArray<any>` -> we get assignment error in callbacks because they're assigning a readonly to a mutable (but this time in reverse)

Reply:
Only way to win here is to correctly type interfaces, which anybody using the standard methods in Tanstack Query will get for free, but anybody passing around query options would need to update their argument types as per the one failing test in this PR

Reply:
Pushed an alternate approach, it's back to where I started but with a clearer user story inspired by tRPC: https://github.com/TanStack/query/pull/8394#discussion_r1881067882